### PR TITLE
Support of unsigned types

### LINKED ISSUE #109: Incorrect binding to unsigned integers
Even though DuckDB supports unsigned integer, this crate still binds values as signed integers, causing overflow error:

```rust
use duckdb::*;
fn main() {
    let conn = Connection::open_in_memory().expect("open db");
    conn.execute_batch(
        r"CREATE TABLE a (x UBIGINT);").expect("create table");
    conn.execute("INSERT INTO a(x) VALUES (?)", params![0xFFFFFFFFFFFFFFFFu64]).expect("insert");
}
```

Output:

```
thread 'main' panicked at 'insert: DuckDBFailure(Error { code: Unknown, extended_code: 1 }, Some("Conversion Error: Type INT64 with value -1 can't be cast because the value is out of range for the destination type UINT64"))', src/main.rs:7:81
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This behavior might be inherited from Rusqlite, because I know Rusqlite does something like this.

### PR DESCRIPTION
**Pull request by rongcuid**
Fixes #109

### CODE REVIEW

**Comment by rongcuid** [comment]
Also fixed appender

File: src/statement.rs [hunk 1]
```diff
@@ -639,6 +643,26 @@ mod test {
             assert!(rows.next()?.is_none());
         }
 
+        {
+            let db = Connection::open_in_memory()?;
+            db.execute_batch("CREATE TABLE test (name TEXT, value UINTEGER)")?;
+            let mut stmt = db.prepare("INSERT INTO test(name, value) VALUES (?, ?)")?;
+            stmt.raw_bind_parameter(1, "negative")?;
+            stmt.raw_bind_parameter(2, 0xFFFFFFFFu32)?;
```
Comment (wangfenjin) [inline src/statement.rs:651]: https://doc.rust-lang.org/std/primitive.u32.html#associatedconstant.MAX

File: src/statement.rs [hunk 2]
```diff
@@ -639,6 +643,26 @@ mod test {
...
+        }
+
+        {
+            let db = Connection::open_in_memory()?;
+            db.execute_batch("CREATE TABLE test (name TEXT, value UBIGINT)")?;
+            let mut stmt = db.prepare("INSERT INTO test(name, value) VALUES (?, ?)")?;
+            stmt.raw_bind_parameter(1, "negative")?;
+            stmt.raw_bind_parameter(2, 0xFFFFFFFFFFFFFFFFu64)?;
+            let n = stmt.raw_execute()?;
+            assert_eq!(n, 1);
```
Comment (wangfenjin) [inline src/statement.rs:663]: Same as u32, please add assert on the value

**Review by wangfenjin — COMMENTED** [review]
Thanks for the contribution. Few minor comments:

**Pull request merged**